### DISCUSSION TITLE
Displaying epics and their user stories, sub-tasks, and chores

### DISCUSSION THREAD

**Discussion by jeff-abrahamson-vectice** [ankitpokhrel/jira-cli#546] [General]
I can find the epics of interest to me with `jira epic list -lMY_TAG --table --plain`.

Is there a way to see the whole tree?
```
TYPE	KEY	SUMMARY					STATUS
Epic	EN-1234	Stuff			                Open
  Story EN-9901 Sub-stuff                               In Progress
  Chore EN-9902 Simple-stuff                            Open
Epic	EN-5678	Other stuff	                        Open
  Story EN-9903 Sub-other-stuff                         In Progress
  Story EN-9904 Other-sub-stuff                         In Progress
```

Maybe there are more sub-tasks, more levels, but this would give me a nice view of everything my team is doing right now.

**Comment by LordReydrX** [comment]
To clarify, this need is a result of not having all the bells and whistles but those insights are needed today?

I am asking because it looks like there is an opportunity to use and issueFunction and recursive links (Depth: 3-4) query. This approach allows you to query from the highest issuetype, and tends to clean up faster than data queried at the middle tier queries.

**Reply by jeff-abrahamson-vectice** [reply]
Not 100% sure I understood your question, but I think you said "what you experience is because some things we might have done in jira-cli aren't done (yet) -- do you need these things today?"

The response to that question is this: if I had it today, it would be really useful to me and I would use it immediately.  But I also realise that getting those things means someone else spending their time on it, so I can only politely say "this would be really useful to me".

**Reply by LordReydrX** [reply]
So, summary: Yes. 

Honestly I have not seen much done with any of the Atlassian products that customers had not been refining for years previously. 

Which tools/resources/practices/Woodoo have you been able to do and how far off of "better" are you now from where you want to be?  

What data is missing?

**Comment by ankitpokhrel** [comment]
Hi @jeff-abrahamson-vectice,

In TUI you can do `jira epic list -lMY_TAG` to view the tree. If you want to use a plain mode then you need to use some scripting to get the tree. Below is a sample script.

```sh
#!/usr/bin/env bash

epics=$(jira epic list -lMY_TAG --table --plain --no-headers --columns TYPE,KEY,SUMMARY,STATUS)

echo "${epics}" | while IFS=$'\t' read -r line; do
    key=$(echo "${line}" | awk '{print $2}')
    issues=$(jira epic list $key --plain --no-headers --columns TYPE,KEY,SUMMARY,STATUS 2>/dev/null)

    printf "%s\n" "${line}"
    echo "${issues}" | while IFS=$'\t' read -r issue; do
        printf "  - %s\n" "${issue}"
    done
done
```

_~P.S. [this comment](https://github.com/ankitpokhrel/jira-cli/discussions/546#discussioncomment-4617749) you replied above is most probably a SPAM~_, my bad

**Comment by jeff-abrahamson-vectice** [comment]
I forgot to respond to this:

* Thanks for the shell script!  That's quite clever and works well.
* I also found [jira-python](https://pypi.org/project/jira/) package, which offers quite a lot of flexibility for more complex tasks, including direct result manipulation in python, but which is quite a bit slower than this go client.
